### PR TITLE
[Android] Implement API getFavicon

### PR DESCRIPTION
**Pull request by shaoboyan**
Implement getFavicon API for users to get favicon on current page.

BUG=XWALK-4919

Signed-off-by: Yan, Shaobo <shaobo.yan@intel.com>

### CODE REVIEW

**Comment by shaoboyan** [comment]
@lincsoon @wuhengzhi  PTAL

File: test/android/core/javatests/src/org/xwalk/core/xwview/test/GetFaviconTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package org.xwalk.core.xwview.test;
+
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.test.suitebuilder.annotation.SmallTest;
+import android.util.Log;
+import android.os.Message;
```
Comment (wuhengzhi) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/GetFaviconTest.java]: alpha sort?

File: test/android/core/javatests/src/org/xwalk/core/xwview/test/GetFaviconTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+import org.chromium.base.test.util.Feature;
+import org.chromium.net.test.util.TestWebServer;
+
+import org.chromium.content.browser.test.util.TestCallbackHelperContainer.OnPageFinishedHelper;
+import org.xwalk.core.XWalkUIClient;
+import org.xwalk.core.XWalkView;
+import org.xwalk.core.XWalkSettings;
```
Comment (wuhengzhi) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/GetFaviconTest.java]: ditto.

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java [hunk 3]
```diff
@@ -1077,6 +1078,17 @@ void setOverlayVideoMode(boolean enabled) {
     }
 
     /**
+     * Return the icon which current page has.
+     */
```
Comment (wuhengzhi) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java]: Miss the API version, explanation for the return value.

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java [hunk 4]
```diff
@@ -1077,6 +1078,17 @@ void setOverlayVideoMode(boolean enabled) {
...
+    @XWalkAPI
+    public Bitmap getFavicon() {
+        if (isOwnerActivityRunning()) {
```
Comment (wuhengzhi) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java]: Please add ```if (mContent == null) return;```

File: test/android/core/javatests/src/org/xwalk/core/xwview/test/GetFaviconTest.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+            loadUrlAsync(pageUrl);
+
+            // The getFavicon will return the right icon a certain time after
+            // the page load completes which makes it slightly hard to test.
+            pollOnUiThread(new Callable<Boolean>() {
+                    @Override
+                    public Boolean call() throws Exception{
+                        return mXWalkView.getFavicon() != null;
+                    }
+                });
```
Comment (wuhengzhi) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/GetFaviconTest.java]: indent?

File: test/android/core/javatests/src/org/xwalk/core/xwview/test/GetFaviconTest.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+        webServer = TestWebServer.start();
+        mXWalkView = getXWalkView();
+        setUIClient(new XWalkUIClient(getXWalkView()) {
+            /**
+             * Must override this function for correct behaviour
+             */
+            @Override
+            public void onIconAvailable(XWalkView view, String url, Message msg) {
+                Log.d("XWalkView", "onIconAvailable");
```
Comment (wuhengzhi) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/GetFaviconTest.java]: This line could be removed.

File: test/android/core/javatests/src/org/xwalk/core/xwview/test/GetFaviconTest.java (hunk 1 above)
Comment (shaoboyan) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/GetFaviconTest.java]: done.

File: test/android/core/javatests/src/org/xwalk/core/xwview/test/GetFaviconTest.java (hunk 2 above)
Comment (shaoboyan) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/GetFaviconTest.java]: done.

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java (hunk 3 above)
Comment (shaoboyan) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java]: done.

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java (hunk 4 above)
Comment (shaoboyan) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewInternal.java]: done.

File: test/android/core/javatests/src/org/xwalk/core/xwview/test/GetFaviconTest.java (hunk 5 above)
Comment (shaoboyan) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/GetFaviconTest.java]: done

File: test/android/core/javatests/src/org/xwalk/core/xwview/test/GetFaviconTest.java (hunk 6 above)
Comment (shaoboyan) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/GetFaviconTest.java]: done

**Comment by xzhan96** [comment]
I'd like to not use "Signed-off-by:", it's the kernel commit standard, but not crosswalk's.

**Comment by shaoboyan** [comment]
@xzhan96  thx for remind. Updated.

File: test/android/core/javatests/src/org/xwalk/core/xwview/test/GetFaviconTest.java [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    private TestWebServer webServer;
+    private XWalkView mXWalkView;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+
+        webServer = TestWebServer.start();
+        mXWalkView = getXWalkView();
+        setUIClient(new XWalkUIClient(getXWalkView()) {
```
Comment (wuhengzhi) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/GetFaviconTest.java]: ```getXWalkView()``` -> ```mXWalkView```

**Comment by wuhengzhi** [comment]
LGTM for others, thanks.

**Comment by shaoboyan** [comment]
@fujunwei PTAL

**Comment by fujunwei** [comment]
LGTM

**Comment by shaoboyan** [comment]
@lincsoon @halton Please help merge this PR. thx.

**Comment by shaoboyan** [comment]
ping @lincsoon  @halton

**Pull request merged**